src/reducers/fileTree.tsx

```diff
@@ -131,20 +132,43 @@ export const buildFileTree = (version: Version): DirectoryNode => {
   return root;
 };
 
+export const buildTreePathList = (tree: DirectoryNode): string[] => {
+  const treePathList: string[] = [];
+
+  const extractPaths = (node: TreeNode) => {
+    if ('children' in node) {
+      for (const child of node.children) {
+        extractPaths(child);
+      }
+    } else {
+      treePathList.push(node.id);
+    }
+  };
+
+  extractPaths(tree);
+
+  return treePathList;
+};
+
 export type FileTreeState = {
   forVersionId: void | number;
   tree: DirectoryNode | void;
+  treePathList: string[] | void;
```

Comment:
Since I suggested doing all the reducer work in response to `buildTreePathList`, it would probably be more convenient to expand the tree object instead of adding a new state attribute. Something like this:

```tsx
tree: {
  nodes: DirectoryNode | void;
  pathList: string[] | void;
}
```

What do you think?

Reply:
Yes, sounds good.